.github/workflows/ci.yml

```
@@ -126,45 +129,196 @@ jobs:
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: arm64
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+            username: ${{ secrets.DOCKER_HUB_USER }}
+            password: ${{ secrets.DOCKER_HUB_PASS }}
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+            registry: ${{ env.GITHUB_REGISTRY }}
+            username: ${{ github.actor }}
+            password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and Push Container
+        uses: docker/build-push-action@v4
+        with:
+            context: backend
+            platforms: linux/amd64,linux/arm64
+            push: true
+            tags: |
+              ${{ env.BACKEND_DOCKER_HUB_IMAGE_NAME }}:${{ github.sha }}-untested,
+              ${{env.GITHUB_REGISTRY}}/${{ env.BACKEND_GITHUB_IMAGE_NAME }}:${{ github.sha }}-untested
+            cache-from: |
+              type=gha,scope=${{github.ref_name}}-backend
+              type=gha,scope=${{github.base_ref}}-backend
+            cache-to: type=gha,mode=max,scope=${{github.ref_name}}-backend
+
+    e2e_ci:
+      name: Lint E2E
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v3
+
+        - name: Install Node.js
+          uses: actions/setup-node@v3
+          with:
+            node-version: 18.x
+
+        - name: Install Dependencies
+          working-directory: e2e
+          run: |
+            npm ci
+
+        - name: Lint
+          working-directory: e2e
+          run: |
+            npm run lint -- --max-warnings 0 --no-fix
+
+    e2e_test:
+      name: Run E2E Tests
+      needs: [e2e_ci,frontend_container,backend_container]
+      runs-on: ubuntu-latest
+      services:
+        backend:
+          image: oeffis/backend:${{ github.sha }}-untested
+          ports:
+            - 3000:3000
+        frontend:
+          image: oeffis/frontend:${{ github.sha }}-untested
+          ports:
+            - 8000:80
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v3
+
+        - name: Install Node.js
+          uses: actions/setup-node@v3
+          with:
+            node-version: 18.x
+
+        - name: Install Dependencies
+          working-directory: e2e
+          run: |
+            npm ci
+
+        - name: Run Tests
+          uses: cypress-io/github-action@v5
+          with:
+            working-directory: e2e
+
+        - name: Upload Artifacts
+          if: always()
+          uses: actions/upload-artifact@v3
+          with:
+            name: cypress-report
+            path: |
+              e2e/cypress/screenshots
+              e2e/cypress/videos
+
+    publish:
+      name: Publish Images
+      needs: [e2e_test]
```

```suggestion
      needs: 
        - e2e_test
```

imo lint should be _needed_ here

---

you mean additionally? I disagree. I would leave the dependencies as they are. Publish doesn't need to know wether the files pass linting, but only wether they run succesfully.